### ISSUE TITLE
[LineChart] Proposal: TooltipItem with Widget child.

### ISSUE THREAD

**Issue by jamesblasco** [imaNNeoFighT/fl_chart#72]
Tooltips in line chart are a really cool feature, but would be even cooler if we could customize it a bit more. 

Making the child a widget become very versatile. It would allow to add a title with a custom style below the spot value or even an image. 

Proposal             |  Now
:-------------------------:|:-------------------------:
<img width="200" alt="Captura de pantalla 2019-09-28 a las 11 54 37" src="https://user-images.githubusercontent.com/19904063/65814817-3f777880-e1e7-11e9-9ca7-4ae3ed73b0d9.png"> | <img width="200" alt="Captura de pantalla 2019-09-28 a las 11 54 37" src="https://user-images.githubusercontent.com/19904063/65814818-40100f00-e1e7-11e9-8c78-b44ea8fd3980.png">

**Comment by imaNNeoFighT** [comment]
Hi, currently the tooltip is drawn by the canvas,
but we can handle it by a widget.
stay tuned!

**Comment by Ntu-odo** [comment]
Hi there ! That feature could really make this package one of the best available, I'm currently looking for this feature for my company, do you have any idea when we could expect this to be released ?

**Comment by DinithHerath** [comment]
Hi, 
Can you implement this feature as soon as possible. Our company also need this.
Thanks for the hard work you are doing.

**Comment by imaNNeoFighT** [comment]
Hi, check out our new release: [0.35.0](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.35.0)

`children` property added  in the [LineTooltipItem](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/line_chart.md#linetooltipitem), [BarTooltipItem](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/bar_chart.md#bartooltipitem) and [ScatterTooltipItem](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/scatter_chart.md#scattertooltipitem) which accepts a list of [TextSpan](https://api.flutter.dev/flutter/painting/TextSpan-class.html). It allows you to have more customized texts inside the tooltip.

Check below examples:

##### BarChartSample 1 ([Source Code](/example/lib/bar_chart/samples/bar_chart_sample1.dart))
<img src="https://github.com/imaNNeoFighT/fl_chart/raw/master/repo_files/images/bar_chart/bar_chart_sample_1.gif" width="300" >

##### ScatterChartSample 2 ([Source Code](/example/lib/scatter_chart/samples/scatter_chart_sample2.dart))
<img src="https://github.com/imaNNeoFighT/fl_chart/raw/master/repo_files/images/scatter_chart/scatter_chart_sample_2.gif" width="300" >

**Comment by WillowWisp** [comment]
@imaNNeoFighT sorry for tagging you in but, the `children` property still doesn't allow us to pass a Widget into the tooltip since it only accepts `List<TextSpan>`

For example, we still cannot achieve a tooltip that looks like this.
![image](https://user-images.githubusercontent.com/37970784/118624339-45becc00-b7f3-11eb-8ea5-ff9bed66b752.png)

Is it possible to change the `children` type from `List<TextSpan>` to `List<InlineSpan>` so that we can pass in a `WidgetSpan`?
Thank you for the cool package btw. It helps a lot!

**Comment by P-B1101** [comment]
Hi and thank you for your great work.
I wonder can u add border for tooltip?

**Comment by imaNNeoFighT** [comment]
@WillowWisp  you're right. It only allows you to put List<TextSpan> in the children.

BTW supporting widgets is an open issue that we are thinking about.
It is related to #183.

I re-open this issue to follow custom widgets.

**Comment by imaNNeoFighT** [comment]
@P-B1101 please search in issues and if there isn't any already opened issue, make a separate issue to follow border on tooltips.

Thanks!

**Comment by eduardothiesen** [comment]
BarChartSample https://github.com/example/lib/bar_chart/samples/bar_chart_sample1.dart is not opening.

**Comment by devasidmi** [comment]
@imaNNeoFighT Hey, greate package! Any updates or workarounds on custom Widget as child for tooltips?

**Comment by demoYang** [comment]
how to custom LineTooltipItem ?

**Comment by noinskit** [comment]
FTR I also needed this feature to draw colored icons over some chart bars and I managed to work it around by putting the icons into a custom font using FontLab (if anybody tries that - it needs to be CBDT, it took me some time to figure out that OT+SVG does not work in Flutter and/or Android 🤷).

**Comment by sgshivamgarg8** [comment]
Any updates?
I want to show Icons or Images over bar rods.

I tried to do this in getTooltipItem but it expects BarTooltipItem which only allows text.

**Comment by biklas7** [comment]
@imaNNeoFighT sorry to tag you, first of all, thanks for the amazing package. Just wanted to ask if this feature is in the roadmap for the next releases?

**Comment by durannumit** [comment]
Hey everyone, I definitely need this feature and everyone waiting on next release. I implemented my custom tooltip a bit hacky way. Firstly,  I created an overlay widget I wrap `BarChart` widget with `Listener` and I'm listening barTouchData which index bar touched. 

```
 Listener(
                        onPointerUp: (event) {
                          overlayEntry?.remove();
                        },
                        onPointerMove: (event) {
                        // show Overlay on the specific index;
                        },
                        onPointerDown: (event) {
                        // show Overlay on the specific index;
                        },
                        child: BarChart(
                          BarChartData(
                            ....
                            barTouchData: BarTouchData(
                              touchCallback: ((p0, p1) {
                                //update touched bar index
                             ....
                               },
                              ),
                            ),
                          ),
                        ),
                      ),
```

I know it's not the best way but it's working and useful for now. Maybe this can be helpful 💪🏻

**Comment by omar-hanafy** [comment]
Yes it is a cool feature to have custom tooltip widget, I can't even add simple elevation to my tooltip 😕.

**Comment by anil-thapa-1** [comment]
@imaNNeo Hi, i am in need of this feature in my project where I need a custom widget as a tooltip. So, an i work in this feature and create a PR?

**Comment by d3xt3r2909** [comment]
Really needed feature..

**Comment by Sumit258000** [comment]
Any plans for this implementations @imaNNeo

**Comment by jebstern** [comment]
Is the maintainer waiting for community to implement this? This issue is over 4 years old.

**Comment by dagovalsusa** [comment]
+1

**Comment by tfitzke** [comment]
I would greatly appreciate it if you could provide information regarding the roadmap for this feature or improvement. Additionally, I am interested in knowing if there are opportunities for community contributions!

**Comment by imaNNeo** [comment]
> I would greatly appreciate it if you could provide information regarding the roadmap for this feature. Additionally, I am interested in knowing if there are opportunities for community contributions!

Hi @tfitzke,
There is no roadmap, unfortunately. This is my side-project and I only contribute in my free-times.
I wish I could work full-time on it, but I can't.
Contributions are always welcome. Check the [guideline](https://github.com/imaNNeo/fl_chart/blob/master/CONTRIBUTING.md) as a starting point.

**Comment by tfitzke** [comment]
> > I would greatly appreciate it if you could provide information regarding the roadmap for this feature. Additionally, I am interested in knowing if there are opportunities for community contributions!
> 
> Hi @tfitzke, There is no roadmap, unfortunately. This is my side-project and I only contribute in my free-times. I wish I could work full-time on it, but I can't. Contributions are always welcome. Check the [guideline](https://github.com/imaNNeo/fl_chart/blob/master/CONTRIBUTING.md) as a starting point.

Hey @imaNNeo 
Thank you for your answer! 
Then I will gladly try to contribute as soon as I can :)

**Comment by egonbeermat** [comment]
Would love if this great package supported adding WidgetSpan to the tooltip TextSpan list for bar charts and other types. Currently throws an error. Solutions to this issue have been discussed here for the same issue in the AutoSizeText package:

[WidgetSpan causes an error](https://github.com/simc/auto_size_text/issues/61)

**Comment by xanderelsmith** [comment]
please has it been done, including inlinespan or widgetspan  to tooltip?

**Comment by VARKOLU1729** [comment]
I wanted to check if there has been any progress on this issue or if someone has already started working on it. 
Please let me know if there's any update.
